// mytoastfy/dist/index.js
// src/toast.ts
function toast(options) {
  injectStyles();
  const {
    message,
    type = "info",
    duration = 3e3,
    position = "top-right"
  } = options;
  const toastElement = document.createElement("div");
  toastElement.className = `my-toast ${type} ${position}`;
  const messageElement = document.createElement("div");
  messageElement.innerText = message;
  const progressBar = document.createElement("div");
  progressBar.className = "my-toast-progress";
  progressBar.style.animationDuration = `${duration}ms`;
  toastElement.appendChild(messageElement);
  toastElement.appendChild(progressBar);
  document.body.appendChild(toastElement);
  setTimeout(() => {
    toastElement.remove();
  }, duration);
}
function injectStyles() {
  if (document.getElementById("mytoastfy-styles")) return;
  const style = document.createElement("style");
  style.id = "mytoastfy-styles";
  style.textContent = `
  .my-toast { 
    

    position: fixed;
    padding: 12px 18px;
    padding-bottom: 16px;
    border-radius: 6px;
    color: white;
    font-family: Arial, sans-serif;
    font-size: 14px;
    min-width: 220px;

    display: flex;
    flex-direction: column;
    text-align: center;
    justify-content: center;
    align-items: center;

    z-index: 9000;
    overflow: hidden;
    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
    animation: fadeIn 0.3s ease-in-out;
}

.my-toast-progress {
    position: absolute;
    bottom: 0;
    left: 0;
    height: 4px;
    width: 100%;
    animation: progress linear forwards;
}

.top-right {
    top: 20px; 
    right: 20px;
}

.top-left {
    top: 20px;
    left: 20px;
}

.bottom-right { 
    bottom: 20px;
    right: 20px;
}

.bottom-left {
    bottom: 20px;
    left: 20px;
}

.center {
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}


.success {
    background-color: #16a34a; 
}

.success .my-toast-progress {
    background-color: #bbf7d0; 
}


.error {
    background-color: #dc2626; 
}

.error .my-toast-progress {
    background-color: #fecaca; 
}


.info {
    background-color: #2563eb; 
}

.info .my-toast-progress {
    background-color: #bfdbfe;
}


.warning {
    background-color: #d97706; 
    color: white;
}

.warning .my-toast-progress {
    background-color: #fde68a; 
}


@keyframes progress {
    from {
        width: 100%;
    }
    to {
        width: 0%;
    }
}

@keyframes fadeIn {
    from {
        opacity: 0;
    }

    to {
        opacity: 1;
    }
}
  `;
  document.head.appendChild(style);
}
export {
  toast
};
